File: app/src/main/java/com/unis/gameplatfrom/adapter/BaseEmptyViewAdapter.java
package com.unis.gameplatfrom.adapter;

import android.content.Context;
import android.widget.LinearLayout;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;


import java.util.ArrayList;
import java.util.List;

import com.unis.gameplatfrom.ui.view.ListEmptyView;

/**
 * Created by wulei on 2016-08-16.
 */

@SuppressWarnings("DefaultFileTemplate")
public abstract class BaseEmptyViewAdapter<T> extends BaseQuickAdapter<T, BaseViewHolder> {

    private boolean showEmpty;

    public BaseEmptyViewAdapter(Context ctx, int layoutResId, List<T> list) {
        this(ctx, true, layoutResId, list);
    }

    public BaseEmptyViewAdapter(Context ctx, boolean showEmpty, int layoutResId, List<T> list) {
        super(layoutResId, list == null ? new ArrayList<T>() : list);
        if (showEmpty) {
            setEmptyMessage(ctx, "暂无数据");
        }
    }

    public void setEmptyMessage(Context ctx, String message) {
        ListEmptyView emptyView = new ListEmptyView(ctx);
        emptyView.setMessage(message);
        emptyView.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT));
        setEmptyView(emptyView);
    }

}

File: kotlin/src/main/java/com/unis/kotlin/api/adapter/BaseEmptyViewAdapter.kt
package com.unis.kotlin.api.adapter

import android.content.Context
import android.widget.LinearLayout
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.unis.kotlin.ui.view.ListEmptyView
import java.util.ArrayList

abstract class BaseEmptyViewAdapter<T> : BaseQuickAdapter<T,BaseViewHolder>{

    constructor(context: Context,layoutResId:Int,list:List<T>):this(context,
            true,layoutResId,list)

    constructor(ctx: Context, showEmpty: Boolean, layoutResId: Int, list: List<T>?) : super(layoutResId,list ?:ArrayList()) {
        if (showEmpty) {
            setEmptyMessage(ctx, "暂无数据")
        }
    }

    fun setEmptyMessage(ctx: Context, message: String) {
        val emptyView = ListEmptyView(ctx)
        emptyView.setMessage(message)
        emptyView.setLayoutParams(LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT))
        setEmptyView(emptyView)
    }

}